### PR TITLE
Check that the kubelet service exists

### PR DESCRIPTION
**Pull request by ereslibre**

### CODE REVIEW

File: provision/minion/provision-salt-minion.sh [hunk 1]
```diff
@@ -34,6 +34,11 @@ while [ $# -gt 0 ] ; do
   shift
 done
 
+does_service_exist() {
+  systemctl list-unit-files | grep $1
+  $?
```
Comment (inercia) [inline provision/minion/provision-salt-minion.sh]: You don't need this `$?`. Just use: 

```
does_service_exist() { systemctl list-unit-files | grep -q $1 &>dev/null ; }
```

**Comment by mjura** [comment]
I hit this same problem, which I reported here https://github.com/kubic-project/terraform/pull/11/. But I will fix it

File: provision/minion/provision-salt-minion.sh (hunk 1 above)
Comment (inercia) [inline provision/minion/provision-salt-minion.sh]: I would also skip the need for `.service`, so I would `grep "$1.service"` and then you could `does_service_exist "kubelet"`

File: provision/minion/provision-salt-minion.sh [hunk 2]
```diff
@@ -73,9 +78,11 @@ mkdir -p /etc/salt/minion.d
 cp -v $SALT_ROOT_SUBDIR/config/minion.d/*  /etc/salt/minion.d
 [ -z $SKIP_ROLE_ASSIGNMENTS ] && cp -v $SALT_ROOT_SUBDIR/grains /etc/salt/
 
-log "Enabling & (re)starting the Salt minion"
-systemctl enable kubelet   || abort "could not enable the kubelet"
-systemctl restart kubelet  || abort "could not restart the kubelet"
+if [ does_service_exist "kubelet.service" ] ; then
```
Comment (inercia) [inline provision/minion/provision-salt-minion.sh]: Are you sure you need the `[]` ? Can't you just `if does_service_exist "kubelet" ; then`

**Comment by ereslibre** [comment]
Thanks @inercia, me and my bash skills...

**Comment by ereslibre** [comment]
Updated, but I'd rather prefer to merge https://github.com/kubic-project/terraform/pull/11 instead of this one.

**Comment by inercia** [comment]
I think this is better than #11 : it is a bit smarter... I will merge.

**Pull request merged**